src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceRegistrationEventListener.cs

```diff
@@ -9,8 +9,16 @@
 
 namespace Microsoft.CodeAnalysis.LanguageServer;
 
-// Specify the VS workspaces we know we need (host, misc, metadata as source) and MSBuild for VSCode.
-[ExportEventListener(WellKnownEventListeners.Workspace, WorkspaceKind.Host, WorkspaceKind.MiscellaneousFiles, WorkspaceKind.MetadataAsSource, WorkspaceKind.MSBuild, WorkspaceKind.Interactive), Shared]
+[ExportEventListener(
+    WellKnownEventListeners.Workspace,
+    WorkspaceKind.Host,
+    WorkspaceKind.MiscellaneousFiles,
+    WorkspaceKind.MetadataAsSource,
+    // TODO(cyrusn): Why does LSP need to know about the msbuild workspace?  It's a workspace that is used in console
+    // apps, not rich server scenarios.
+    WorkspaceKind.MSBuild,
+    // TODO(cyrusn): Why does LSP need to know about the interactive workspace? Does LSP work in interactive buffers?
```

Comment:
It does , for example when the all lsp flag is on, it powers completion in interactive.

Reply:
fascinating.